src/main/java/com/javadeobfuscator/deobfuscator/transformers/special/RadonTransformerV2.java

```diff
@@ -110,6 +122,116 @@ public boolean canCheckEquality(JavaValue first, JavaValue second, Context conte
+						continue;
+
+					Map<Integer, AtomicInteger> insnCount = new HashMap<>();
+					Map<String, AtomicInteger> invokeCount = new HashMap<>();
+					for (AbstractInsnNode i = m.instructions.getFirst(); i != null; i = i.getNext()) {
+						int opcode = i.getOpcode();
+						insnCount.putIfAbsent(opcode, new AtomicInteger(0));
+						insnCount.get(opcode).getAndIncrement();
+						if (i instanceof MethodInsnNode) {
+							invokeCount.putIfAbsent(((MethodInsnNode) i).name, new AtomicInteger(0));
+							invokeCount.get(((MethodInsnNode) i).name).getAndIncrement();
+						}
+					}
+
+					if (insnCount.get(Opcodes.NEWARRAY) == null || insnCount.get(Opcodes.ISTORE) == null
+							|| insnCount.get(Opcodes.BALOAD) == null || insnCount.get(Opcodes.IOR) == null
+							|| invokeCount.get("toCharArray") == null || invokeCount.get("getResourceAsStream") == null
+							|| invokeCount.get("getMethodName") == null)
+						continue;
+
+					decr = m;
+					ownerdecr = classNode.name;
+				}
+			}
+
+			if (decr != null)
+				for (ClassNode classNode : classNodes()) {
+					for (MethodNode m : classNode.methods) {
+						InstructionModifier modifier = new InstructionModifier();
+						for (AbstractInsnNode ain : TransformerHelper.instructionIterator(m))
+							if (ain instanceof MethodInsnNode) {
+								MethodInsnNode mi = (MethodInsnNode) ain;
+
+								if (mi.owner.equals(ownerdecr) && mi.name.equals(decr.name)) {
+									AbstractInsnNode ldc = ain.getPrevious();
+									if (!(ldc instanceof LdcInsnNode) || !(((LdcInsnNode) ldc).cst instanceof String))
+										continue;
+									Context context = new Context(provider);
+									context.push(classNode.name, m.name,
+											getDeobfuscator().getConstantPool(classNode).getSize());
+									context.dictionary = classpath;
+
+									for (AbstractInsnNode i = decr.instructions.getFirst(); i != null; i = i
+											.getNext()) {
+										if (i.getOpcode() == Opcodes.INVOKEVIRTUAL
+												&& ((MethodInsnNode) i).owner.equals("java/lang/Class")
+												&& ((MethodInsnNode) i).name.equals("getResourceAsStream")
+												&& ((MethodInsnNode) i).desc
+														.equals("(Ljava/lang/String;)Ljava/io/InputStream;"))
+											MethodExecutor.customMethodFunc.put(i, (list, ctx) -> {
+
+												try {
+													ZipFile zipIn = new ZipFile(
+															getDeobfuscator().getConfig().getInput());
+													InputStream in = zipIn
+															.getInputStream(zipIn.getEntry(classNode.name + ".class"));
+
+													ByteArrayOutputStream baos = CloneInputStream(in);
+
+													InputStream stream1 = new ByteArrayInputStream(baos.toByteArray());
+													return JavaValue.valueOf(stream1);
+												} catch (IOException e1) {
+													e1.printStackTrace();
+												}
+
+												return null;
+											});
+										else if (i.getOpcode() == Opcodes.INVOKEVIRTUAL
```

Comment:
Instead of this, you can use the Context#push() methods to spoof the callstack without the need to define a hooked function behaviour.